File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/BalanceCommand.java
package ro.razvan.java.dagger.atm;

import javax.inject.Inject;

public class BalanceCommand implements NoArgsCommand {

    public static final String COMMAND_KEY_BALANCE = "balance";

    private final Outputter outputter;
    private final Database.Account account;

    @Inject
    public BalanceCommand(Outputter outputter, Database.Account account) {
        this.outputter = outputter;
        this.account = account;
    }

    @Override
    public Result handle() {
        outputter.output("Your balance is: " + account.balance());
        return Result.handled();
    }

    @Override
    public String key() {
        return COMMAND_KEY_BALANCE;
    }
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/BalanceCommand.kt
package ro.razvan.kotlin.dagger.atm

import javax.inject.Inject

const val COMMAND_KEY_BALANCE = "balance"

class BalanceCommand @Inject constructor(
    private val account: Database.Account,
    private val outputter: Outputter
) : NoArgsCommand {

    override fun handle(): Command.Result {
        outputter("Your balance is: ${account.balance}")
        return Command.Result.handled()
    }

    override val key: String
        get() = COMMAND_KEY_BALANCE
}